polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -137,16 +140,29 @@ export function Item({
       <div className={styles.Badge}>{badgeMarkup}</div>
     );
 
+  const tone =
+    !showVerticalHoverPointer && !matches && level !== 0
+      ? 'subdued'
+      : undefined;
+  let fontWeight: TextProps['fontWeight'] = 'regular';
+  if (matches && !childIsActive) {
```

Comment:
```suggestion
 if ((matches && !childIsActive) || (level === 0 && selected)) {
```
Selected L1 items also need `semibold` font weight. Comparing selected Products nav item on [prod](https://storybook.polaris.shopify.com/?path=/story/all-components-navigation--with-an-active-root-item-with-secondary-navigation-items)/[storybook](https://5d559397bae39100201eedc1-yrjhwxlxrv.chromatic.com/?path=/story/all-components-navigation--with-an-active-root-item-with-secondary-navigation-items).

Reply:
Are you sure? When looking at the prod storybook, looks like **Products** is `medium` and the active child item is `semibold`.

<img width="144" alt="21-26-ndob4-e43u2" src="https://github.com/Shopify/polaris/assets/11774595/f7ce5335-50c8-4f73-92e7-e3603648b41b">
<img width="125" alt="21-32-p6wfx-we3uc" src="https://github.com/Shopify/polaris/assets/11774595/f8717a0a-9b2b-48cd-ad31-f3ddade5e95c">

Reply:
Sorry, my terminology was a bit mixed up because we've referred to `level=0` items as L1 for view transition work 😅

There is a discrepancy in the story I linked earlier between this branch's storybook and prod. Home and Products are both active by default and none of the subnavs for Prod are active. On this branch, Home still has `semibold` applied but Products has `medium`.
<img width="300" alt="nav-pr" src="https://github.com/Shopify/polaris/assets/26749317/fc66562e-b1c4-455a-9b65-21df913f7d13">

Whereas in prod, Home has `semibold` because it's active but because Products is active too, it also has `semibold`. 
<img width="300" alt="nav-prod" src="https://github.com/Shopify/polaris/assets/26749317/8543ad46-4592-4598-84c1-9a587da4e9d6">

I actually think the logic in my suggestion can be simplified after testing the story locally and comparing it to prod. I don't think the level matters, we just need to also check that `selected` is true to apply `semibold`
```suggestion
  if ((matches || selected) && !childIsActive) {
```

Reply:
Ah, I see. Good catch, this is an odd state imo where we have two actively selected nav items 🤔 but best to keep the behavior the same. Thanks, Lo!